packages/@vue/cli-plugin-eslint/generator/index.js

```diff
@@ -2,49 +2,75 @@ const fs = require('fs')
 const path = require('path')
 
 module.exports = (api, { config, lintOn = [] }, _, invoking) => {
-  if (typeof lintOn === 'string') {
-    lintOn = lintOn.split(',')
-  }
-
   const eslintConfig = require('../eslintOptions').config(api)
+  const extentions = require('../eslintOptions').extensions(api)
+    .map(ext => ext.replace(/^\./, ''))  // remove the leading `.`
 
   const pkg = {
     scripts: {
       lint: 'vue-cli-service lint'
     },
     eslintConfig,
     devDependencies: {
-      'eslint': '^5.16.0',
-      'eslint-plugin-vue': '^5.0.0'
+      eslint: '^6.7.2',
+      'eslint-plugin-vue': '^6.0.1'
     }
   }
 
   if (!api.hasPlugin('typescript')) {
+    // required for dynamic import support
     pkg.devDependencies['babel-eslint'] = '^10.0.3'
   }
 
-  if (config === 'airbnb') {
+  switch (config) {
+  case 'airbnb':
     eslintConfig.extends.push('@vue/airbnb')
     Object.assign(pkg.devDependencies, {
-      '@vue/eslint-config-airbnb': '^4.0.0'
+      '@vue/eslint-config-airbnb': '^5.0.0',
+      'eslint-plugin-import': '^2.18.2'
     })
-  } else if (config === 'standard') {
+    break
+  case 'standard':
     eslintConfig.extends.push('@vue/standard')
     Object.assign(pkg.devDependencies, {
-      '@vue/eslint-config-standard': '^4.0.0'
+      '@vue/eslint-config-standard': '^5.0.0',
+      'eslint-plugin-import': '^2.18.2',
+      'eslint-plugin-node': '^9.1.0',
+      'eslint-plugin-promise': '^4.2.1',
+      'eslint-plugin-standard': '^4.0.0'
     })
-  } else if (config === 'prettier') {
-    eslintConfig.extends.push('@vue/prettier')
+    break
+  case 'prettier':
+    eslintConfig.extends.push(
+      ...(api.hasPlugin('typescript')
+        ? ['eslint:recommended', '@vue/typescript/recommended', '@vue/prettier', '@vue/prettier/@typescript-eslint']
+        : ['eslint:recommended', '@vue/prettier']
+      )
+    )
     Object.assign(pkg.devDependencies, {
-      '@vue/eslint-config-prettier': '^5.0.0',
+      '@vue/eslint-config-prettier': '^6.0.0',
       'eslint-plugin-prettier': '^3.1.1',
       prettier: '^1.19.1'
     })
-    // prettier & default config do not have any style rules
-    // so no need to generate an editorconfig file
-  } else {
+    break
+  default:
     // default
     eslintConfig.extends.push('eslint:recommended')
+    break
+  }
+
+  // typescript support
+  if (api.hasPlugin('typescript')) {
+    Object.assign(pkg.devDependencies, {
+      '@vue/eslint-config-typescript': '^5.0.1',
+      '@typescript-eslint/eslint-plugin': '^2.10.0',
+      '@typescript-eslint/parser': '^2.10.0'
+    })
+    if (config !== 'prettier') {
+      // for any config other than `prettier`,
+      // typescript ruleset should be appended to the end of the `extends` array
+      eslintConfig.extends.push('@vue/typescript/recommended')
```

Comment:
Not sure whether we should use `@vue/typescript/recommended` or `@vue/typescript` here.